cmd/thanos/receive.go

```diff
@@ -233,16 +233,15 @@ func runReceive(
 
 	level.Debug(logger).Log("msg", "setting up receive http handler")
 	{
-		ctx, cancel := context.WithCancel(context.Background())
 		g.Add(
 			func() error {
-				if err := webHandler.Run(ctx); err != nil {
+				if err := webHandler.Run(); err != nil {
 					return fmt.Errorf("error starting web server: %s", err)
```

Comment:
`errors.Wrap` instead, also I am 90% sure `Run` will never return nil error (: Not a blocker.